evm_arithmetization/tests/log_opcode.rs

```diff
@@ -573,23 +624,81 @@ fn test_log_with_aggreg() -> anyhow::Result<()> {
     };
 
     let mut timing = TimingTree::new("prove root second", log::Level::Info);
-    let (root_proof_second, public_values_second) =
-        all_circuits.prove_root(&all_stark, &config, inputs, &mut timing, None.clone())?;
+    let max_cpu_len = 1 << 20;
+    let root_proof_data_second = all_circuits.prove_segment(
+        &all_stark,
+        &config,
+        inputs.clone(),
+        max_cpu_len,
+        0,
+        &mut timing,
+        None.clone(),
+    )?;
+    let ProverOutputData {
+        proof_with_pis: root_proof_second,
+        public_values: public_values_second,
+    } = root_proof_data_second;
     timing.filter(Duration::from_millis(100)).print();
 
     all_circuits.verify_root(root_proof_second.clone())?;
 
-    let (agg_proof, updated_agg_public_values) = all_circuits.prove_aggregation(
-        false,
-        &root_proof_first,
-        public_values_first,
-        false,
-        &root_proof_second,
-        public_values_second,
+    let final_root_proof_data_second = all_circuits.prove_segment(
+        &all_stark,
+        &config,
+        inputs,
+        max_cpu_len,
+        1,
+        &mut timing,
+        None.clone(),
+    )?;
```

Comment:
I'm confused. Why do we need a second segment? the `max_cpu_len` is 1M.

Reply:
This is for aggregation: `prove_segment_aggregation` requires two segments.